### ISSUE TITLE
cache issue | notification on event card

### ISSUE THREAD

**Issue by Symbiot78** [gcobb321/icloud3_v3#234]
Hi 

to me this is a bit peculiar:

I've added icoud3. I have NOT added the event log card.
Randomly after reboots.. but not after every reboot ... I'll get a notification in HA from icloud3 telling me that the eventlog card has been added to HA and that I should clear my cache. I check my resources in HA - and no event log card. 
The notification reads:
######
Action Required - Clear Browser Cache
The Event Log Custom Card was added to the Lovelace Resource list.
File-icloud3-event-log-card.js

The browser cache must be cleared before the Event Log Custom Card can be added.
1. Press Ctrl+Shift+Del
2. On the Settings tab, check Clear Images and Files
3. Click Clear Data/Clear Now
4. Select the Home Assistant tab, then Refresh the display

iCloud3 Notification 2023-12-01 06:48:11
#####

Not sure what device the above instructions are meant for.

ok - to the actual problem : whenever the above notification appears my dashboards break. almost all cards are shows as "resource not found" (not sure about wording).
It happens in the HA companion app on mobile (iOS)
On my mobile I can reset frontend cache in the companion app.. but it doesn't help. I can do it as many times as i like - no difference.

After some time, som reboots, some 'clear cache' som waiting.. .. it'll suddenly go back to working.

If I disable icloud3 integration or remove it completely - the issue goes away.

When I refresh dashboard in iOS companion app I see this error in ha log:

`Logger: homeassistant.components.websocket_api.http.connection
Source: components/lovelace/websocket.py:68
Integration: Home Assistant WebSocket API (documentation, issues)
First occurred: 08:59:38 (2 occurrences)
Last logged: 09:02:59

[140199495948992] Error handling message: Unknown error (unknown_error) martin from 192.168.1.231 (Mozilla/5.0 (iPhone; CPU iPhone OS 17_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Home Assistant/2023.7 (io.robbie.HomeAssistant; build:2023.471; iOS 17.2.0) Mobile/HomeAssistant, like Safari)
[140199537534912] Error handling message: Unknown error (unknown_error) martin from 192.168.1.231 (Mozilla/5.0 (iPhone; CPU iPhone OS 17_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Home Assistant/2023.7 (io.robbie.HomeAssistant; build:2023.471; iOS 17.2.0) Mobile/HomeAssistant, like Safari)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/decorators.py", line 26, in _handle_async_response
    await func(hass, connection, msg)
  File "/usr/src/homeassistant/homeassistant/components/lovelace/websocket.py", line 68, in websocket_lovelace_resources
    await resources.async_load()
          ^^^^^^^^^^^^^^^^^^^^
AttributeError: 'ResourceYAMLCollection' object has no attribute 'async_load'
`

**Comment by Symbiot78** [comment]
oh - it also happens on laptop btw.

**Comment by gcobb321** [comment]
1. iCloud3 version=? Latest iCloud3 v3.0.rc8
2. Are you planning on adding the _Event Log_? Without it, trying to see what it’s going on with the interaction between iCloud3, iCloud and the iosapp is almost impossible. 
3. The iCloud3 docs explain about the _Event Log_ and the resource setup for it. iCloud3 tries to update the resources but might fail to do so in some configuration. See docs [here](https://gcobb321.github.io/icloud3_v3_docs/#/chapters/1.1-ic3-components?id=event-log) and [here](https://gcobb321.github.io/icloud3_v3_docs/#/chapters/2.0-installing-and-configuring?id=step-53-create-the-event-log-portion-of-the-lovelace-screen) and [here](https://gcobb321.github.io/icloud3_v3_docs/#/chapters/1.4-evlog-during-startup)

Now to the problem. It sounds like iCloud3 is scanning the HA resources, which it does, and doesn’t find it so it is trying to add it. For some reason, that is failing in your system. Try adding it manually so iCloud3 will find that entry, even if you are not going to use it. The browser you are using (Chrome, Safari, etc) to view the HA Lovelace cards is what the message refers to.

**Comment by Symbiot78** [comment]
Hi 
Thanks for answering.

I created a blank dashboard to add the card. Using UI wasn't possible. It didn't find the card. So I 'just' added your example in the docs manually and changed the entities to my own phone.
Everything shows up in the card(s) - So now I guess I'll need to wait.. test.. see.

My version is v3.0.rc8.1 - to answer your question.

**Comment by strickersuk** [comment]
I'm having similar issues, I follow the reset frontend cache instructions on the app but the event log is stuck at 3.0.12 and not 3.0.16. The instructions in the app don't update the event log. In Safari the instructions say press CTL+ALT+DEL, there is no DEL option on a Mac keyboard?

**Comment by gcobb321** [comment]
@strickersuk 
See the iCloud3 docs [here](https://gcobb321.github.io/icloud3_v3_docs/#/chapters/3.1-migrating-v2-to-v3?id=step-5-clear-the-browsers-cache) on clearing the browser cache. it’s CTL-SHIFT-DEL, not CTL-ALT-DEL. You might have to do this several times. 

Google clearing cache if you need to. 
You can also verify the version that Is actually loading:
- Check the Lovelace resource and make sure that directory matches the Event Log directory  on the _Configure Settings > Update Parameters Menu > Other Tracking Parameters_ screen
- Go to that directory, open the _icloud3-event-log-card.js_file and look at the version number around Line 25.

**Comment by strickersuk** [comment]
Thanks Gary, I meant CTL-SHIFT-DEL. There is no DEL key on a Mac Keyboard, only backspace. The loaded version is 3.0.16 yet all my instances are saying 3.0.12 and tell me to reset cache. I've done this several times on the app but the warning is still there?

**Comment by gcobb321** [comment]
Removing it will not solve anything. That it’s telling you that you are running iCloud3 rc9 (the correct version), that _Event Log_ 3.0.16 is installed on the HA www directory (the correct version) and 3.0.12 is what is loaded into the browser memory and actually running. The problem is the browsers cache or the HA cache. That needs to be cleared so 3.0.16 will load. 

Power off your iPhone if you have to. Then power it back on and see what loads.

**Comment by strickersuk** [comment]
OK, we got there in the end. Looks like we had more than one instance of the event log card on my HA, I tidied that, double checked the resource then followed the reset instructions for the app, this worked. I then loaded into Firefox which I don't normally use and this showed 3.0.16. 

To reset the cache in Safari (No Del Key), I followed this process:

Safari>Settings>Advanced>Show features for web developers

Then:

Develop>Empty Caches

Thanks for the prompt replies Gary.
